tx/transaction.go

```diff
@@ -42,46 +50,66 @@ type Transaction struct {
 	}
 }
 
-// body describes details of a tx.
-type body struct {
-	ChainTag     byte
-	BlockRef     uint64
-	Expiration   uint32
-	Clauses      []*Clause
-	GasPriceCoef uint8
-	Gas          uint64
-	DependsOn    *thor.Bytes32 `rlp:"nil"`
-	Nonce        uint64
-	Reserved     reserved
-	Signature    []byte
+// TxData describes details of a tx.
+type TxData interface {
+	txType() byte
+	copy() TxData
+
+	chainTag() byte
+	blockRef() uint64
+	expiration() uint32
+	clauses() []*Clause
+	gasPriceCoef() uint8
```

Comment:
Is the `gasPriceCoef` still required in the tx body ?

Reply:
We still need it as part of the legacy tx body for backward compatibility. We could remove it from the `TxData` interface and replace it with `maxFeePerGas` (so when calling `tx.maxFeePerGas()` on a legacy tx it would return the `gasPriceCoef`); but imo that would make things less clear.

Reply:
Seems this is how geth does it too